packages/manager/apps/web-hosting/src/data/hooks/webHostingAttachedDomain/useWebHostingAttachedDomain.ts

```diff
@@ -44,23 +48,25 @@ export const useWebHostingAttachedDomain = (
       ),
   });
 
-  const fetchAllPages = useCallback(() => {
-    if (!allPages) {
-      setAllPages(true);
-    }
-  }, [allPages, setAllPages]);
-  useEffect(() => {
-    if (allPages && query.hasNextPage && !query.isFetchingNextPage) {
-      query.fetchNextPage();
-    }
-  }, [query.data]);
-  useEffect(() => {
-    if (!shouldFetchAll) {
-      setAllPages(false);
-    }
-  }, []);
+  const allDataQuery = useQuery<ApiResponse, Error>({
```

Comment:
If you only use this call only for your csv export, you don't have to put it in here or use `useQuery`, you could just make a simple function that call `getWebHostingAttachedDomain` until all results are fetched